### ISSUE TITLE
Latex doesn't render in table of contents

### ISSUE THREAD

**Issue by somanath08** [squidfunk/mkdocs-material#1610]
__I checked that...__

- [x] ... the documentation does not mention anything about my problem
- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... there are no open or closed issues that are related to my problem

## Description

When I use the latex in my headings, the corresponding entry for the heading in the table of contents doesn't render the latex properly.

Here is an image, which I believe would make my problem more apparent.

![image](https://user-images.githubusercontent.com/15904169/79530880-4fc14900-808e-11ea-85fc-b50609903fcb.png)

### Expected behaviour

The latex used in headings should also show up in the table of contents.

### Package versions

* Python: `3.6.10`
* MkDocs: `1.1`
* Material: `5.0.2`

### Project configuration

Here are the extensions I used

``` yaml
# Extensions
markdown_extensions:
  - markdown.extensions.admonition
  - markdown.extensions.attr_list
  - markdown.extensions.codehilite
  - markdown.extensions.def_list
  - markdown.extensions.meta
  - markdown.extensions.toc:
      permalink: '#'
  - pymdownx.arithmatex
  - pymdownx.betterem:
      smart_enable: all
  - pymdownx.caret
  - pymdownx.critic
  - pymdownx.details
  - pymdownx.inlinehilite
  - pymdownx.keys
  - pymdownx.magiclink:
      repo_url_shorthand: true
      user: somanath08
      repo: Telugu-Analysis
  - pymdownx.mark
  - pymdownx.smartsymbols
  - pymdownx.snippets:
      check_paths: true
  - pymdownx.superfences
  - pymdownx.tabbed
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.tilde

# For MathJax
extra_javascript:
  - https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML
```

### System information

* OS: macOS Catalina, 10.15.3
* Browser: Google Chrome, Version 80.0.3987.132

**Comment by facelessuser** [comment]
Parsing is not a theme issue. With that said, there is an issue over at Python Markdown that covers this: https://github.com/Python-Markdown/markdown/issues/895.

**Comment by squidfunk** [comment]
I cannot stress enough how important it is to actually try to reproduce the error with the default MkDocs theme. It's so simple - just comment out the whole theme section and see if the issue still occurs. It would save us maintainers so much valuable time.

With that said, closing as upstream issue.

**Comment by somanath08** [comment]
Sorry about that, the default MKDocs theme doesn't render the third-level heading and that was why I wasn't able to reproduce the error. I should have checked that with a second-level heading or something. 

*I did check it with the default theme.* I should probably have mentioned that too. My bad.  Sorry for any trouble I have caused you guys. 

@facelessuser thank you for pointing me to a reference.
